test/test_runner_battle.c

```diff
@@ -1672,13 +2052,157 @@ void ForcedMove(u32 sourceLine, struct BattlePokemon *battler)
     }
 }
 
+static void TryMarkExpectedMove(u32 sourceLine, struct BattlePokemon *battler, struct MoveContext *ctx)
+{
+    s32 battlerId = battler - gBattleMons;
+    u32 moveId, moveSlot, id;
+    s32 target;
+
+    INVALID_IF(DATA.turnState == TURN_CLOSED, "EXPECTED_MOVE outside TURN");
```

Comment:
I'd add an `INVALID_IF(!IsAITest(), ...);` to this and the other AI-only expected functions.